mailet/standard/src/test/java/org/apache/james/transport/mailets/StripAttachmentTest.java

```diff
@@ -155,201 +191,530 @@ public void testSimpleAttachment2() throws MessagingException, IOException {
+    @Test
+    public void analyseMultipartPartMessageShouldReturnFalseWhenNoPartHasBeenRemovedInSubMultipart() throws Exception {
+        //Given
+        StripAttachment mailet = new StripAttachment();
 
-        recover.service(mail);
+        FakeMailetConfig mci = FakeMailetConfig.builder()
+                .mailetName("Test")
+                .setProperty("remove", "matched")
+                .setProperty("pattern", ".*\\.tmp")
+                .build();
+        mailet.init(mci);
+        MimeMultipart mainMultiPart = new MimeMultipart();
+        MimeMessage subMessage = mimeMessage();
+        MimeMultipart subMultiPart = new MimeMultipart();
+        MimeBodyPart part = new MimeBodyPart(new ByteArrayInputStream(new byte[0]));
+        part.setFileName("dontRemoveMe.other");
+        subMultiPart.addBodyPart(part);
+        subMessage.setContent(subMultiPart);
+        subMessage.saveChanges();
+        mainMultiPart.addBodyPart(new MimeBodyPart(new InternetHeaders(new ByteArrayInputStream("Content-Type: multipart/mixed".getBytes(Charsets.US_ASCII)))
+                ,
+                IOUtils.toByteArray(subMessage.getInputStream())));
+        MimeMessage message = mimeMessage();
+        message.setContent(mainMultiPart);
+        message.saveChanges();
+        Mail mail = mock(Mail.class);
+        //When
+        boolean actual = mailet.analyseMultipartPartMessage(message, mail);
+        //Then
+        assertThat(actual).isFalse();
+    }
 
-        Assert.assertTrue(mail.getMessage().getContent() instanceof MimeMultipart);
-        Assert.assertEquals(2, ((MimeMultipart) mail.getMessage().getContent())
-                .getCount());
+    @Test
+    public void analyseMultipartPartMessageShouldRemovePartWhenOnePartShouldHaveBeenRemoved() throws Exception {
+        //Given
+        StripAttachment mailet = new StripAttachment();
 
-        Object actual = ((MimeMultipart) mail.getMessage().getContent())
-                .getBodyPart(1).getContent();
-        if (actual instanceof ByteArrayInputStream) {
-            Assert.assertEquals(body2, toString((ByteArrayInputStream) actual));
-        } else {
-            Assert.assertEquals(body2, actual);
-        }
+        FakeMailetConfig mci = FakeMailetConfig.builder()
+                .mailetName("Test")
+                .setProperty("remove", "all")
+                .setProperty("pattern", ".*")
+                .build();
+        mailet.init(mci);
+        MimeMultipart multiPart = new MimeMultipart();
+        
+        MimeBodyPart part = new MimeBodyPart(new ByteArrayInputStream(new byte[0]));
+        part.setFileName("removeMe.tmp");
+        multiPart.addBodyPart(part);
+        MimeMessage message = mimeMessage();
+        message.setContent(multiPart);
+        message.saveChanges();
+        Mail mail = mock(Mail.class);
+        //When
+        mailet.analyseMultipartPartMessage(message, mail);
+        //Then
+        assertThat(multiPart.getCount()).isZero();
```

Comment:
so analyse is doing more than analysing, it has some side effects. Please rename it.